design-toolkit/components/src/components/kanban/index.tsx

```diff
@@ -0,0 +1,422 @@
+    setActiveId(event.active.id as string);
+  };
+
+  const handleDragEnd = (event: DragEndEvent) => {
+    setActiveId(null);
+
+    if (!moveCard) {
+      return;
+    }
+
+    const dropTarget = parseDropTarget(event);
+    if (!dropTarget) {
+      return;
+    }
+
+    moveCard(
+      event.active.id as string,
+      dropTarget.columnId,
+      dropTarget.position,
+      dropTarget.edge,
+    );
+  };
+
+  // Find the active card for the drag overlay
+  const activeCard = activeId
+    ? columns.flatMap((col) => col.cards).find((card) => card.id === activeId)
+    : null;
+
+  return (
+    <DragContext.Provider value={{ activeId }}>
+      <DndContext
+        sensors={sensors}
+        collisionDetection={collisionDetectionStrategy}
+        onDragStart={handleDragStart}
+        onDragEnd={handleDragEnd}
+      >
+        <div className={container({ className })} {...rest}>
+          {children}
+        </div>
+        <DragOverlay>
+          {activeCard ? (
+            <div className={CardInnerStyles({ isActive: true })}>
+              <div className={cardHeader()}>
+                <span className={cardTitle()}>{activeCard.title}</span>
+              </div>
+              <div className={cardBody()}>{activeCard.body}</div>
+            </div>
+          ) : null}
+        </DragOverlay>
+      </DndContext>
+    </DragContext.Provider>
+  );
+}
+Kanban.displayName = 'Kanban.Container';
+
+const Header = ({ children, className, ...rest }: KanbanComponentProps) => {
+  return (
+    <div className={header({ className })} {...rest}>
+      {children}
+    </div>
+  );
+};
+Header.displayName = 'Kanban.Header';
+
+const HeaderTitle = ({
+  children,
+  className,
+  ...rest
+}: KanbanComponentProps) => {
+  return (
+    <Heading className={headerTitle({ className })} {...rest}>
+      {children}
+    </Heading>
+  );
+};
+HeaderTitle.displayName = 'Kanban.Header.Title';
+
+const HeaderActions = ({
+  children,
+  className,
+  ...rest
+}: KanbanComponentProps) => (
+  <div className={headerActions({ className })} {...rest}>
+    {children}
+  </div>
+);
+HeaderActions.displayName = 'Kanban.Header.Actions';
+
+const HeaderSearch = ({ classNames, ...rest }: KanbanSearchProps) => (
+  <SearchField
+    variant='outlined'
+    classNames={classNames}
+    aria-label='Search'
+    {...rest}
+  />
+);
+HeaderSearch.displayName = 'Kanban.Header.Search';
+
+const ColContainer = ({
+  children,
+  className,
+  ...rest
+}: KanbanComponentProps) => {
+  return (
+    <div className={colContainer({ className })} {...rest}>
+      {children}
+    </div>
+  );
+};
+ColContainer.displayName = 'Kanban.Column.Container';
+
+const Col = ({ children, className, column, ...rest }: KanbanColProps) => {
+  const { ref, isActive, isHighlighted } = useColumnInteractions(column);
+
+  return (
+    <div
+      className={ColumnStyles({ isHighlighted, isActive, className })}
+      ref={ref}
+      {...rest}
```

Comment:
nit: could add ARIA roles of list or region for columns and listitem for cards.
priority: low

```suggestion
      role='list'
      {...rest}
```